src/Forms/Shared/Samples/Location/IndoorPositioning/IndoorPositioning.xaml.cs

```diff
@@ -0,0 +1,204 @@
+                    {
+                        await Application.Current.MainPage.DisplayAlert("Error", "Location permission required for use of indoor positioning.", "OK");
+                        return;
+                    }
+                }
+                catch
+                {
+                    return;
+                }
+
+#if __ANDROID__
+                // Get bluetooth permission for Android devices. Devies running Android 12 or higher need the `BluetoothScan` permission. Android 11 and below require the `Bluetooth` and `BluetoothAdmin` permissions.
+                bool bluetoothScanGranted = await MainActivity.Instance.AskForBluetoothPermission();
+                if (!bluetoothScanGranted)
+                {
+                    await Application.Current.MainPage.DisplayAlert("Error", "Bluetooth permission is required for use of indoor positioning.", "OK");
+                    return;
+                }
+#endif
+
+                // Create a portal item for the web map.
+                ArcGISPortal portal = await ArcGISPortal.CreateAsync(_portalUri, true);
+                PortalItem item = await PortalItem.CreateAsync(portal, ItemId);
+
+                // Load the map in the map view.
+                MyMapView.Map = new Map(item);
+                await MyMapView.Map.LoadAsync();
+
+                // Get the positioning table from the map.
+                await Task.WhenAll(MyMapView.Map.Tables.Select(table => table.LoadAsync()));
+                FeatureTable positioningTable = MyMapView.Map.Tables.FirstOrDefault(table => table.TableName.Equals(PositioningTableName));
+                if (positioningTable == null) return;
+
+                // Get a table of all of the indoor pathways.
+                FeatureLayer pathwaysFeatureLayer = MyMapView.Map.OperationalLayers.FirstOrDefault(l => l.Name.Equals(PathwaysLayerName, StringComparison.InvariantCultureIgnoreCase)) as FeatureLayer;
+                ArcGISFeatureTable pathwaysTable = pathwaysFeatureLayer.FeatureTable as ArcGISFeatureTable;
+
+                // Get the global id for positioning.
+                Field dateCreatedFieldName = positioningTable.Fields.FirstOrDefault(f => f.Name.Equals("DateCreated", StringComparison.InvariantCultureIgnoreCase) || f.Name.Equals("DATE_CREATED", StringComparison.InvariantCultureIgnoreCase));
+
+                QueryParameters queryParameters = new QueryParameters
+                {
+                    MaxFeatures = 1,
+                    // "1=1" will give all the features from the table.
+                    WhereClause = "1=1",
+                };
+                queryParameters.OrderByFields.Add(new OrderBy(dateCreatedFieldName.Name, SortOrder.Descending));
+
+                FeatureQueryResult queryResult = await positioningTable.QueryFeaturesAsync(queryParameters);
+                Guid globalID = (Guid)queryResult.First().Attributes["GlobalID"];
```

Comment:
It's not clear to me why the sample is doing this. Could we explain why we're just making a query and grabbing whatever guid comes back first? Will the datasource not work without this guid?